keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -591,44 +592,45 @@ class SqlArtifactRepository(
       return false
     }
 
-    return selectPromotionReference(envUid, artUid, veto.version)
-      .fetchOne { (ref: String?) ->
-        ref?.let { reference ->
-          /**
-           * If there's a promotion reference, that means this artifact version was deployed as a result of
-           * another artifact version being vetoed. In that case, we don't veto unless [force] is enabled.
-           */
-          if (!force) {
-            log.warn(
-              "Not vetoing artifact version as it appears to have already been an automated rollback target: " +
-                "deliveryConfig=${deliveryConfig.name}, " +
-                "environment=${veto.targetEnvironment}, " +
-                "artifactVersion=${veto.version}, " +
-                "priorVersionReference=$reference")
-            return@fetchOne false
-          }
-        }
+    /**
+     * If there's a promotion reference, that means this artifact version was deployed as a result of
+     * another artifact version being vetoed. In that case, we don't veto unless [force] is enabled.
+     */
+    selectPromotionReference(envUid, artUid, veto.version)
+      .fetchOne() // null if [veto.version] never made it to the target environment
+      ?.value1() // null if the `promotion_reference` column is NULL
+      ?.let { reference ->
```

Comment:
I _think_ you could rewrite this as follows to make it easier to understand:
```suggestion
      .fetchOne(ENVIRONMENT_ARTIFACT_VERSION.REFERENCE)
      ?.let { reference ->
```

Reply:
Nice! TIL. 

The suggested change isn't *quite* right (the name of the field is `PROMOTION_REFERENCE`), but I'll apply the suggested change in a commit.

Reply:
Fixed in f05ef40f1